kratos/geometries/geometry.h

```diff
@@ -727,7 +727,9 @@ class Geometry : public PointerVector<TPointType>
 
         array_1d<double, 3> normal;
         MathUtils<double>::CrossProduct(normal, tangent_xi, tangent_eta);
-        return normal/norm_2(normal);
+	const double norm_normal = norm_2(normal);
+	if (norm_normal > 0.0) normal /= norm_normal;
+        return normal;
```

Comment:
you should throw an error if the norm_normal < eps

Reply:
In debug or always?